### PR TITLE
feat: migrate to new inspector API

### LINKED ISSUE #2: Migrate to new revm API
Once stable, migrate to revm main which includes several breaking changes:

Inspector API calls are no longer recursive this means we need to maintain our own call stack

### PR DESCRIPTION
**Pull request by mattsse**
Closes #2

Blocked by reth migration

and 

https://github.com/bluealloy/revm/issues/1001

### CODE REVIEW

**Comment by rakita** [comment]
@mattsse didn't you need the call/create inputs in Inspector `*_end` functions? I made solution for it here: https://github.com/bluealloy/revm/pull/1003

**Comment by mattsse** [comment]
I think we only need this for the foundry cheatcodes inspector,
here this is fine

File: src/stack/mod.rs [hunk 1]
```diff
@@ -15,7 +17,7 @@ pub use maybe_owned::MaybeOwnedInspector;
 /// - Block: Hook on block execution
 /// - BlockWithIndex: Hook on block execution transaction index
 /// - Transaction: Hook on a specific transaction hash
```
[the remark this answers is not in the document]
Comment (mattsse) [reply src/stack/mod.rs:19]: yeah, this was the case previously, tbh I don't know why this is here.
Perhpas the idea was to move the stack from foundry to here, but I'd rather delete this one here, will probably do so.

**Comment by mattsse** [comment]
lgtm

File: Cargo.toml [hunk 2]
```diff
@@ -4,7 +4,7 @@ description = "Revm inspector implementations"
 version = "0.1.0"
 edition = "2021"
 rust-version = "1.74.0"
-license = "MIT OR Apache-2.0"
+license = "MIT or Apache-2.0"
```
Comment (DaniPopes) [inline Cargo.toml:7]: Why?

Comment (rakita) [reply Cargo.toml:7]: was getting a lint error, will revert

**Comment by rakita** [comment]
will merge after CI

**Pull request merged**